AppController/lib/app_dashboard.rb

```diff
@@ -48,6 +47,17 @@ module AppDashboard
   #   true if the AppDashboard was started successfully, and false otherwise.
   def self.start(login_ip, uaserver_ip, public_ip, private_ip,
       persistent_storage, secret)
+
+    # Pass the secret key and our public IP address (needed to connect to the
+    # AppController) to the app.
+    Djinn.log_run("cp #{APPSCALE_HOME}/AppDashboard/lib/secret_key.py /tmp/secret_key.py")
+    Djinn.log_run("cp #{APPSCALE_HOME}/AppDashboard/lib/local_host.py /tmp/local_host.py")
+    Djinn.log_run("cp #{APPSCALE_HOME}/AppDashboard/lib/uaserver_host.py /tmp/uaserver_host.py")
```

Comment:
The code was using app_location that is defined just below the 3 lines:
if you move them after the definition of app_location, there is no
hardcoded path for those templates.


On Fri, 2016-06-10 at 16:45 -0700, Meni Vaitsi wrote:
> In AppController/lib/app_dashboard.rb:
> > @@ -48,6 +47,17 @@ module AppDashboard
> >    #   true if the AppDashboard was started successfully, and false
> otherwise.
> >    def self.start(login_ip, uaserver_ip, public_ip, private_ip,
> >        persistent_storage, secret)
> > +
> > +    # Pass the secret key and our public IP address (needed to
> connect to the
> > +    # AppController) to the app.
> > +    Djinn.log_run("cp
> #{APPSCALE_HOME}/AppDashboard/lib/secret_key.py /tmp/secret_key.py")
> > +    Djinn.log_run("cp
> #{APPSCALE_HOME}/AppDashboard/lib/local_host.py /tmp/local_host.py")
> > +    Djinn.log_run("cp
> #{APPSCALE_HOME}/AppDashboard/lib/uaserver_host.py
> /tmp/uaserver_host.py")
> The value of these constants is the same in all three templates.
> We can even just overwrite them with a constant after we're done
> tarring up the app dir.
> Is that ok?
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.